Modules/Core/Common/include/itkSobelOperator.hxx

```diff
@@ -76,7 +76,9 @@ template <typename TPixel, unsigned int VDimension, typename TAllocator>
 auto
 SobelOperator<TPixel, VDimension, TAllocator>::GenerateCoefficients() -> CoefficientVector
 {
-  std::vector<double> coeff;
+  CoefficientVector     coeff;
+  constexpr std::size_t coeff_size = (VDimension == 2) ? 9 : 27;
+  coeff.reserve(coeff_size);
```

Comment:
The commit message does not mention the `reserve`, is it a significant performance improvement?

Reply:
I guess `coeff_size` happens to be 3^N with N = `VDimension`, right? If so, you might consider `coeff_size = itk::Math::UnsignedPower(3, VDimension)`.

Reply:
probably not significant given only 9 or 27 elements.  I forgot that I did that.  I think that the current code better documents the intended behaviors.

Reply:
I added the comment, but I don't want to add an non-constexpr computation for the size until the Sobel operator works for more than 2D and 3D that it is currently hard coded for.

Reply:
`Math::UnsignedPower(3, VDimension)` _is_ constexpr  😃 But anyway, thanks!

BTW, if the Sobel  coefficients remain hard coded, it would be more efficient to directly return an initializer list of those coefficients, instead of storing them in a local variable:

    if (VDimension == 2 && this->GetDirection() == 0)
    {
      // No need to reserve or push_back   :-)
      return { -1.0, 0.0, 0.0, 1.0, 1.0, -2.0, -2.0, 0.0, 0.0, 2.0, 2.0, -1.0, -1.0, 0.0, 0.0, 1.0, 1.0 };
    }

But that's probably beyond this PR, no problem!